### DISCUSSION TITLE
Server Action stream timing out while using `streamObject` from Vercel AI SDK (NextJs)

### DISCUSSION THREAD

**Discussion by WillGarman** [vercel/community#6751] [Help] [answered]
## Problem
I may be missing something, but is there any reason my NextJS server action is timing out in the middle of streaming a response back to the client.

Everything runs fine locally (of course), but when deployed to Vercel, I will get part way through the stream and get an error in the console, and the stream stops

```
23-ca6bf042f399153f.js:2 Uncaught (in promise) Error: Connection closed.
```
```
Vercel: Task timed out after 10.01 seconds
```
## Code:
I am using the `Stream Object Generation` example from [Vercel AI SDK](https://sdk.vercel.ai/examples/next-app/basics/streaming-object-generation#server) 
I am also using the client code [also from the docs](https://sdk.vercel.ai/examples/next-app/basics/streaming-object-generation#client)

here is where I handle the client side result (minor change from the docs to check if returned JSON is valid)
```
const { object } = await generate(result);

for await (const partialObject of readStreamableValue(object)) {
if (partialObject) {
  const result = schema.safeParse(partialObject);
  if (result.success) {
    setGeneration(result.data.questions);
  }
}
```

## Video showing stream stopping (in the middle of streaming in the JSON)
![chrome_kV7ejT0iSl](https://github.com/vercel/community/assets/73144742/1a8d2ea6-dba1-443a-aef1-63e3b40160d8)

### Steps to Reproduce

Setup a new NextJS App

Copy the server & client code from the docs 
[https://sdk.vercel.ai/examples/next-app/basics/streaming-object-generation](https://sdk.vercel.ai/examples/next-app/basics/streaming-object-generation)

Update to a prompt that is long and takes a while for the AI to think about
```
const { partialObjectStream } = await streamObject({
  model: openai("gpt-3.5-turbo"),
  system: "You generate 10 questions for a studying app.",
  prompt: `
    Generate 10 questions with 4 multiple choice answers based off of the content.
    Each question should be intresting, basic, and somewhat fun. Think about how DuoLingo asks questions.
    Return as a json object, which has the following format:
    [{"question": "What is the capital of France?","answers": ["New York", "Paris", "London", "Berlin"],"correct_answer": "Paris", "explanation": "'quote' shows xyz"}, ...]
    
    Begin Content:
    ${input}
    End Content`,
  schema: z.object({
    questions: z.array(
      z.object({
        question: z.string(),
        answers: z.array(z.string()),
        correct_answer: z.string(),
        explanation: z.string(),
      })
    ),
  }),
});
```

Deploy to vercel - load the site - hits the 10 seconds timeout limit
```
Vercel logs: Task timed out after 10.01 seconds
```

**Comment by threefoldo** [comment]
yes, I have the same error, but 15s timeout.

**Reply by leerob** [reply, marked as answer]
Hey there, are you using the latest version of Next.js? And have you set a max duration in your page or API?

- https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#maxduration

**Comment by WillGarman** [comment]
> Hey there, are you using the latest version of Next.js? And have you set a max duration in your page or API?
> 
> * https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#maxduration

This did the trick, Thanks Lee! -

If anyone else sees this: I'm using a server action (exported into a different file) so I just added the following line to the `page.tsx` where I use the action
```
export const maxDuration = 30; // Allow server actions 30 seconds to execute
```

**Reply by threefoldo** [reply]
Yes, it works but with problems. I change the duration to 60 seconds, the execution will not be interrupted again, but the streaming mechanism does not work, users have to wait about 40 seconds to get the result.

I tried this method, didn't work:
https://sdk.vercel.ai/docs/troubleshooting/common-issues/streaming-not-working-in-production

**Reply by WillGarman** [reply]
Interesting, sounds like it is a different issue than mine since streaming was working for me it was just timing out. 

But if you post your server and client side code it might be a little easier for other to help you troubleshoot. 

Is this what your code is based on? When I had copy and pasted the server and client code from here I didn't have any issues streaming 
[https://sdk.vercel.ai/examples/next-app/basics/streaming-object-generation#server](https://sdk.vercel.ai/examples/next-app/basics/streaming-object-generation#server)

**Reply by threefoldo** [reply]
Yes, it works fine on local environment, but after deployed to vercel, the streaming is not working. Streaming is essential to user experience, I'm thinking to break it down into small pieces. Right now, I try to generate several sections and put each of them into one edit box. Maybe it's better to generate one chunk of text for each edit box.
